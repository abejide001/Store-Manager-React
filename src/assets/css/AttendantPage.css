@import url("https://fonts.googleapis.com/css?family=Charmonman|Roboto|K2D");
/* product area */

#product-area {
  min-height: 600px;
  font-family: "K2D", sans-serif;
}

#product-area h2 {
  text-align: center;
  font-size: 40px;
}

.product {
  float: left;
  width: 30%;
  padding: 10px;
  text-align: center;
  overflow: hidden;
  border: 1px solid gray;
}

.product-image {
  height: 150px;
  cursor: pointer;
  transition: all 1s ease;
}

.product-name,
.product-price {
  font-weight: bold;
}

.product-image:hover {
  transform: scale(1.2);
}

.product-button {
  background: gray;
  padding: 1rem 2rem;
  color: white;
  border: 0px;
  cursor: pointer;
  outline: none;
  font-size: 12px;
  font-weight: bold;
}

.cart-section {
  min-height: 300px;
}

#cart-title {
  text-align: center;
  font-size: 40px;
  font-family: "K2D", sans-serif;
}

/* cart area */

.cart-header {
  font-weight: bold;
  font-size: 1.25em;
  color: #333;
}

.cart-column {
  display: flex;
  align-items: center;
  border-bottom: 2px solid black;
  margin: 10px 1.5em 0px 0px;
  padding-bottom: 10px;
  font-family: "K2D", sans-serif;
}

.cart-row {
  display: flex;
}

.cart-item {
  width: 45%;
}

.cart-price {
  width: 20%;
  font-size: 1.2em;
  color: #333;
}

.cart-quantity {
  width: 35%;
}

.cart-item-title {
  color: #333;
  margin-left: 0.5em;
  font-size: 1.2em;
}

.cart-item-image {
  width: 75px;
  height: auto;
  border-radius: 10px;
}

.btn-danger {
  height: 30px;
  background: red;
  padding: 5px;
  margin: 12px 0px;
  cursor: pointer;
  font-size: 18px;
  font-weight: bold;
  outline: none;
  color: white;
  font-family: "K2D", sans-serif;
}

.cart-quantity-input {
  height: 34px;
  width: 50px;
  border-radius: 5px;
  border: 1px solid #56ccf2;
  background-color: #eee;
  color: #333;
  padding: 0;
  text-align: center;
  font-size: 1.2em;
  margin-right: 25px;
}

.cart-row:last-child {
  border-bottom: 1px solid black;
}

.cart-row:last-child .cart-column {
  border: none;
}

.cart-total {
  text-align: end;
  margin-top: 10px;
  margin-right: 10px;
}

.cart-total-title {
  font-weight: bold;
  font-size: 1.5em;
  font-family: "K2D", sans-serif;
  color: black;
  margin-right: 20px;
}

.cart-total-price {
  color: #333;
  font-size: 20px;
  font-family: "K2D", sans-serif;
}

.purchase-button {
  display: block;
  margin: 40px auto 80px auto;
  background: gray;
  /* height: 30px; */
  padding: 1rem 2rem;
  color: white;
  border: 0px;
  cursor: pointer;
  outline: none;
  font-size: 12px;
  font-weight: bold;
}

input {
	outline: none;
}
input[type=search] {
	-webkit-appearance: textfield;
	-webkit-box-sizing: content-box;
	font-family: inherit;
	font-size: 100%;
}



input[type=search] {
	background: #ededed url(https://static.tumblr.com/ftv85bp/MIXmud4tx/search-icon.png) no-repeat 9px center;
	border: solid 1px #ccc;
	padding: 9px 10px 9px 32px;
	width: 55px;
	
	-webkit-border-radius: 10em;
	-moz-border-radius: 10em;
	border-radius: 10em;
	
	-webkit-transition: all .5s;
	-moz-transition: all .5s;
	transition: all .5s;
}
input[type=search]:focus {
	width: 130px;
	background-color: #fff;
	border-color: #66CC75;
	
	-webkit-box-shadow: 0 0 5px rgba(109,207,246,.5);
	-moz-box-shadow: 0 0 5px rgba(109,207,246,.5);
	box-shadow: 0 0 5px rgba(109,207,246,.5);
}


input:-moz-placeholder {
	color: #999;
}
input::-webkit-input-placeholder {
	color: #999;
}


@media (max-width: 348px) {
  .products .product {
    float: none;
    width: 100%;
    border: none;
  }
  .cart-section {
    width: 100%;
    float: none;
    text-align: center;
  }
  .cart-row {
    overflow: auto;
  }
}
@media (max-width: 898px) {
  .product {
    float: none;
    width: 100%;
    border: none;
  }
  header #header-title,
    header nav,
    header nav li {
      float: none;
      width: 100%;
      text-align: center;
    }
    header a {
      text-align: center;
      font-size: 17px;
    }
}