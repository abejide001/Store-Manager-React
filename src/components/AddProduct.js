/* eslint-disable camelcase */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { addProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AddProduct.css';

export class AddProduct extends Component {
  state = {
    name: '',
    price: '',
    quantity_in_inventory: '',
    product_image: '',
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    const productData = {
      name,
      price,
      quantity_in_inventory,
      product_image,
    };
    this.props.addProducts(productData);
  };

  handleChange = (e) => {
    this.setState({ [e.currentTarget.name]: e.currentTarget.value });
  };

  render() {
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="/admin">Admin</Link>
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="create-product">
          <form className="submission-form" onSubmit={this.handleSubmit}>
            <h1>Create Product</h1>
            <input
              type="text"
              name="name"
              placeholder="Product Name"
              value={name}
              onChange={this.handleChange}
              required
            />
            <input
              type="number"
              name="price"
              placeholder="Price"
              onChange={this.handleChange}
              value={price}
              required
            />
            <input
              type="number"
              name="quantity_in_inventory"
              value={quantity_in_inventory}
              onChange={this.handleChange}
              placeholder="quantity in inventory"
              required
            />
            <input
              type="url"
              name="product_image"
              value={product_image}
              placeholder="Image URL"
              onChange={this.handleChange}
              required
            />
            <input type="submit" value="CREATE PRODUCT" id="sendBtn" />
          </form>
        </section>
      </div>
    );
  }
}
AddProduct.propTypes = {
  addProducts: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { addProducts })(AddProduct);
