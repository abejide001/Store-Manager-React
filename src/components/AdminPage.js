/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts, deleteProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/Admin.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

handleDelete = (id) => {
  this.props.deleteProduct(id);
}

render() {
  const { products } = this.props.product.products;
  return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/create-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    <button className="editBtn"><Link to={`edit-product?id=${item.id}`}>Edit</Link></button>
                    {' '}
                    <button className="deleteBtn" type="submit" onClick={() => this.handleDelete(item.id)}>Delete</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts, deleteProduct })(Admin);
