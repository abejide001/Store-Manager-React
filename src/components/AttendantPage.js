import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import Footer from './layouts/Footer';
import logo from '../assets/images/logo.png';
import '../assets/css/AttendantPage.css';
import '../assets/css/style.css';
import Spinner from '../common/Spinner';

export class AttendantPage extends Component {
  state = {
    search: '',
  };

  componentDidMount() {
    this.props.getProducts();
  }

  handleChange = (e) => {
    this.setState({ search: e.target.value });
  };

  render() {
    const { products } = this.props.product.products;
    const { search } = this.state;
    const foundProducts = products
      && products.value.filter(
        product => product.name.toLowerCase().includes(search.toLowerCase()),
      );

    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="./available-products">Available Products</Link>
                </li>
                <li>
                  <Link to="./individual-attendant-sale-record">Records</Link>
                </li>
                <li>
                  <input
                    type="search"
                    placeholder="Search Product"
                    value={search}
                    name="search"
                    onChange={this.handleChange}
                  />
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="product-area">
          <div className="container">
            <h2>Products</h2>
            <div className="products">
              {foundProducts === undefined ? (
                <Spinner />
              ) : (
                foundProducts.map(item => (
                  <div className="product-item" key={item.id}>
                    <img src={item.product_image} className="product-image" />
                    <p className="product-name">{item.name}</p>
                    <p className="product-price">
#
                      {(item.price).toLocaleString()}
                                        </p>
                    <p>
                      <button className="product-button" type="button">
                        ADD TO CART
                      </button>
                    </p>
                  </div>
                ))
              )}
            </div>
          </div>
        </section>
        <section className="container cart-section">
          <div className="container">
            <h1 id="cart-title">CART</h1>
            <div className="cart-row">
              <span className="cart-item cart-header cart-column">ITEM</span>
              <span className="cart-price cart-header cart-column">PRICE</span>
              <span className="cart-quantity cart-header cart-column">
                QUANTITY
              </span>
            </div>
            <div className="cart-items" />
            <div className="cart-total">
              <strong className="cart-total-title">Total</strong>
              <span className="cart-total-price">#0</span>
            </div>
            <button className="purchase-button" type="button">
              PURCHASE
            </button>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(
  mapStateToProps,
  { getProducts },
)(AttendantPage);
