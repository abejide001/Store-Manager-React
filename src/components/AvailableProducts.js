/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AvailableProducts.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

  render() {
    const { products } = this.props.product.products;
    return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/add-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Available Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    {' '}
                    <button className="modalBtn" type="submit">Details</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts })(Admin);
