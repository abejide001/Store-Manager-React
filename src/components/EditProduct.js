/* eslint-disable react/forbid-prop-types */
/* eslint-disable no-undef */
/* eslint-disable camelcase */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Footer from './layouts/Footer';
import { editProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/EditProduct.css';

export class EditProduct extends Component {
    state = {
      name: '',
      price: '',
      quantity_in_inventory: '',
      product_image: '',
    }

    handleSubmit = (event) => {
      event.preventDefault();
      const regex = /\d+/g;
      const id = this.props.location.search.match(regex);
      const { name, price, quantity_in_inventory, product_image } = this.state;
      const productData = {
        name,
        price,
        quantity_in_inventory,
        product_image,
      };
      this.props.editProduct(id, productData);
    }

    handleChange = (e) => {
      this.setState({ [e.currentTarget.name]: e.currentTarget.value });
    };

    render() {
      const {
        name, price, quantity_in_inventory, product_image,
      } = this.state;
      return (
        <div>
          <div className="edit-body">
            <div className="edit-showcase">
              <header>
                <div className="container">
                  <div id="header-title">
                    <Link to="/">
                      <img src={logo} alt="logo" />
                    </Link>
                  </div>
                  <nav>
                    <ul>
                      <li>
                        <Link to="/admin">Admin</Link>
                      </li>
                    </ul>
                  </nav>
                </div>
              </header>
              <section id="edit">
                <form className="submission-form" onSubmit={this.handleSubmit}>
                  <h1>EDIT PRODUCT</h1>
                  <input type="text" className="input1" name="name" placeholder="Product Name" value={name} onChange={this.handleChange} required />
                  <input type="number" className="input2" name="price" placeholder="Price" value={price} onChange={this.handleChange} required />
                  <input type="number" className="input3" name="quantity_in_inventory" placeholder="quanity in inventory" value={quantity_in_inventory} onChange={this.handleChange} required />
                  <input type="url" name="product_image" placeholder="Image URL" value={product_image} onChange={this.handleChange} required />
                  <input type="submit" value="EDIT" id="sendBtn" />
                </form>
              </section>
              <Footer />
            </div>
          </div>
        </div>
      );
    }
}
EditProduct.propTypes = {
  location: PropTypes.object.isRequired,
  editProduct: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { editProduct })(EditProduct);
