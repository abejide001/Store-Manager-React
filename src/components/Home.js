import React from 'react';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Header from './layouts/Header';
import HomeBody from './layouts/HomeBody';

export default function Home() {
  return (
    <div className="landing-page">
      <Header />
      <HomeBody />
      <Footer />
    </div>
  );
}
