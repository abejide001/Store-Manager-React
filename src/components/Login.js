/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-unused-prop-types */
/* eslint-disable import/no-webpack-loader-syntax */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable comma-dangle */
/* eslint-disable react/no-access-state-in-setstate */
import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-unresolved
import '!style-loader!css-loader!react-toastify/dist/ReactToastify.css';
import '../assets/css/Login.css';
import '../assets/css/style.css';
import logo from '../assets/images/logo.png';
import Footer from './layouts/Footer';
import { loginUser } from '../actions/authActions';

export class Login extends Component {
  state = {
    data: {
      email: '',
      password: '',
    },
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      email: this.state.data.email,
      password: this.state.data.password,
    };
    this.props.loginUser(user, this.props.history);
  }

handleChange = (e) => {
  const data = { ...this.state.data };
  data[e.currentTarget.name] = e.currentTarget.value;
  this.setState({ data });
};

render() {
  const { email, password } = this.state.data;
  return (
    <div>
      <div className="login-body">
        <div className="login-showcase">
          <header>
            <div className="container">
              <div id="header-title">
                <Link to="/">
                  <img src={logo} alt="logo" />
                </Link>
              </div>
            </div>
          </header>
          <section id="login-area">
            <div className="container">
              <form onSubmit={this.handleSubmit}>
                <div className="login-box">
                  <h1>Login</h1>
                  <div className="login-textbox">
                    <i className="fas fa-envelope" />
                    <input type="email" placeholder="Email" name="email" onChange={this.handleChange} value={email} required />
                  </div>
                  <div className="login-textbox">
                    <i className="fas fa-key" aria-hidden="true" />
                    <input type="password" placeholder="Password" name="password" onChange={this.handleChange} value={password} required />
                  </div>
                  <button type="submit" className="btn">{this.props.auth.isLogging ? 'LOGGING IN...' : 'LOGIN'}</button>
                </div>
              </form>
            </div>
          </section>
        </div>
        <Footer />
      </div>
    </div>
  );
}
}
Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  auth: PropTypes.object.isRequired,
  error: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};
const mapStateToProps = state => ({
  auth: state.auth,
  error: state.error,
});
export default connect(mapStateToProps, { loginUser })(withRouter(Login));
