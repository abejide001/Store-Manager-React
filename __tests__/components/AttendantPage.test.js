/* eslint-disable import/no-duplicates */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import props from '../__mocks__/attendantMock';
import { AttendantPage } from '../../src/components/AttendantPage';

let wrapper;
describe('Snapshot', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<AttendantPage {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Name of the group', () => {
    let getProducts;
    beforeEach(() => {
      getProducts = sinon.stub(props, 'getProducts');
    });
    afterEach(() => {
      getProducts.reset();
    });
    it('should call component did moubt', () => {
      wrapper = shallow(<AttendantPage {...props} />);
      expect(getProducts.calledOnce).toBe(true);
    });
  });
});
