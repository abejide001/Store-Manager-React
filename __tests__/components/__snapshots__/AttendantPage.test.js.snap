// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should match snapshot 1`] = `
<div>
  <header>
    <div
      className="container"
    >
      <div
        id="header-title"
      >
        <Link
          replace={false}
          to="/"
        >
          <img
            alt="logo"
            src="test-file-stub"
          />
        </Link>
      </div>
      <nav>
        <ul>
          <li>
            <Link
              replace={false}
              to="./available-products"
            >
              Available Products
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="./individual-attendant-sale-record"
            >
              Records
            </Link>
          </li>
          <li>
            <input
              name="search"
              onChange={[Function]}
              placeholder="Search Product"
              type="search"
              value=""
            />
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <section
    id="product-area"
  >
    <div
      className="container"
    >
      <h2>
        Products
      </h2>
      <div
        className="products"
      />
    </div>
  </section>
  <section
    className="container cart-section"
  >
    <div
      className="container"
    >
      <h1
        id="cart-title"
      >
        CART
      </h1>
      <div
        className="cart-row"
      >
        <span
          className="cart-item cart-header cart-column"
        >
          ITEM
        </span>
        <span
          className="cart-price cart-header cart-column"
        >
          PRICE
        </span>
        <span
          className="cart-quantity cart-header cart-column"
        >
          QUANTITY
        </span>
      </div>
      <div
        className="cart-items"
      />
      <div
        className="cart-total"
      >
        <strong
          className="cart-total-title"
        >
          Total
        </strong>
        <span
          className="cart-total-price"
        >
          #0
        </span>
      </div>
      <button
        className="purchase-button"
        type="button"
      >
        PURCHASE
      </button>
    </div>
  </section>
  <Footer />
</div>
`;
